### PR TITLE
Add has_child query

### PR DESCRIPTION
**Pull request by kolchfa-aws**
Add has_child query

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _query-dsl/joining/has-child.md [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+          "parent_doc": "child_doc"
+        }
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+In this example, you'll configure an index contains documents representing products and their brands.
```
Comment (Naarcha-AWS) [inline _query-dsl/joining/has-child.md:36]:
```suggestion
In this example, you'll configure an index that contains documents representing products and their brands. 
```

**Review by Naarcha-AWS — APPROVED** [review]
One small suggestion, but LGTM.

File: _query-dsl/joining/has-child.md [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+| `max_children` | Optional | The maximum number of matching child documents for a parent document. If exceeded, the parent document is excluded from the search results. |
+| `min_children` | Optional | The minimum number of matching child documents required for a parent document to be included in the results. If not met, the parent is excluded. Default is `1`.|
+| `score_mode` | Optional | Defines how scores of matching child documents influence the parent document's score. Valid values are: <br> - `none`: Ignores the relevance scores of child documents and assigns a score of `0` to the parent document. <br> - `avg`: Uses the average relevance score of all matching child documents. <br> - `max`: Assigns the highest relevance score from the matching child documents to the parent. <br> - `min`: Assigns the lowest relevance score from the matching child documents to the parent. <br> - `sum`: Sums up the relevance scores of all matching child documents. <br> Default is `none`. |
```
Comment (natebower) [inline _query-dsl/joining/has-child.md:190]:
```suggestion
| `score_mode` | Optional | Defines how scores of matching child documents influence the parent document's score. Valid values are: <br> - `none`: Ignores the relevance scores of child documents and assigns a score of `0` to the parent document. <br> - `avg`: Uses the average relevance score of all matching child documents. <br> - `max`: Assigns the highest relevance score from the matching child documents to the parent. <br> - `min`: Assigns the lowest relevance score from the matching child documents to the parent. <br> - `sum`: Sums the relevance scores of all matching child documents. <br> Default is `none`. |
```

File: _query-dsl/joining/has-child.md [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+
+
+## Sorting limitations
+
+The `has_child` query does not support [sorting results]({{site.url}}{{site.baseurl}}/search-plugins/searching-data/sort/) using standard sorting options. If you need to sort parent documents by fields in their child documents, you can use a [`function_score` query]({{site.url}}{{site.baseurl}}/query-dsl/compound/function-score/) and sort by the parent document’s score.
```
Comment (natebower) [inline _query-dsl/joining/has-child.md:195]:
```suggestion
The `has_child` query does not support [sorting results]({{site.url}}{{site.baseurl}}/search-plugins/searching-data/sort/) using standard sorting options. If you need to sort parent documents by fields in their child documents, you can use a [`function_score` query]({{site.url}}{{site.baseurl}}/query-dsl/compound/function-score/) and sort by the parent document's score. 
```

File: _query-dsl/joining/has-child.md [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+  "sales_count": 100,
+  "product_to_brand": {
+    "name": "product", 
+    "parent": "2" 
+  }
+}
+```
+{% include copy-curl.html %}
+
+To search for the parent of a child, use a `has_child` query. The following query returns parent documents (brands) that make watches:
```
Comment (natebower) [inline _query-dsl/joining/has-child.md:118]:
```suggestion
To search for the parent of a child, use a `has_child` query. The following query returns parent documents (brands) that sell watches:
```

File: _query-dsl/joining/has-child.md [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+
+
+## Sorting limitations
+
+The `has_child` query does not support [sorting results]({{site.url}}{{site.baseurl}}/search-plugins/searching-data/sort/) using standard sorting options. If you need to sort parent documents by fields in their child documents, you can use a [`function_score` query]({{site.url}}{{site.baseurl}}/query-dsl/compound/function-score/) and sort by the parent document’s score. 
+
+For the preceding example, you can sort parent documents (brands) based on the `sales_count` of their child products. This query multiplies the score by the `sales_count` field of the child documents and assigns the highest relevance score from the matching child documents to the parent:
```
Comment (natebower) [inline _query-dsl/joining/has-child.md:197]:
```suggestion
In the preceding example, you can sort parent documents (brands) based on the `sales_count` of their child products. This query multiplies the score by the `sales_count` field of the child documents and assigns the highest relevance score from the matching child documents to the parent:
```

**Review by natebower — COMMENTED** [review]
@kolchfa-aws Please see my comments and changes and let me know if you have any questions. Thanks!

File: _field-types/supported-field-types/join.md [hunk 6]
```diff
@@ -61,7 +61,7 @@ PUT testindex1/_doc/1
 ```
 {% include copy-curl.html %}
 
-When indexing child documents, you have to specify the `routing` query parameter because parent and child documents in the same relation have to be indexed on the same shard. Each child document refers to its parent's ID in the `parent` field.
+When indexing child documents, you have to specify the `routing` query parameter because parent and child documents in the same relation have to be indexed on the same shard. For more information, see [Routing]({{site.url}}{{site.baseurl}}/field-types/metadata-fields/routing/). Each child document refers to its parent's ID in the `parent` field.
```
[the remark this answers is not in the document]
Comment (kolchfa-aws) [reply _field-types/supported-field-types/join.md:65]: This means parent/child relation so I can remove.

**Pull request merged**